datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1610,6 +1625,93 @@ fn build_statistics_expr(
     Ok(statistics_expr)
 }
 
+fn extract_string_literal(expr: &Arc<dyn PhysicalExpr>) -> Result<&String> {
+    if let Some(lit) = expr.as_any().downcast_ref::<phys_expr::Literal>() {
+        if let ScalarValue::Utf8(Some(s)) = lit.value() {
```

Comment:
I think you should probably also handle the cases `ScalarValue::LargeUtf8`, `ScalarValue::Utff8View` as well.

Reply:
And `Dictionary`!